codegen/test-utils/src/main/java/software/amazon/smithy/java/codegen/test/PluginTestRunner.java

```diff
@@ -0,0 +1,258 @@
+        }
+        return result;
+    }
+
+    /**
+     * Removes single line comments that can be added to the expected class to annotate specific behavior.
+     */
+    private static String removeSingleLineComments(String contents) {
+        return Arrays.asList(contents.split("\\n"))
+                .stream()
+                .filter(NOT_COMMENT)
+                .collect(Collectors.joining("\n"))
+                .trim();
+    }
+
+    public static TestCaseBuilder builder() {
+        return new TestCaseBuilder();
+    }
+
+    public static class TestCase {
+        private final String name;
+        private final SmithyBuild builder;
+        private final List<MockManifest> manifests;
+        private final Map<String, String> expectedToContents;
+
+        TestCase(TestCaseBuilder builder) {
+            this.name = Objects.requireNonNull(builder.name, "name");
+            this.builder = Objects.requireNonNull(builder.builder, "builder");
+            this.manifests = Objects.requireNonNull(builder.manifests, "manifest");
+            this.expectedToContents = Objects.requireNonNull(builder.expectedToContents, "expectedToContents");
+        }
+
+        public String name() {
+            return name;
+        }
+
+        public SmithyBuild builder() {
+            return builder;
+        }
+
+        public List<MockManifest> manifests() {
+            return manifests;
+        }
+
+        public Map<String, String> expectedToContents() {
+            return expectedToContents;
+        }
+
+        @Override
+        public String toString() {
+            return name;
+        }
+    }
+
+    public static class TestCaseBuilder {
+        private String name;
+        private SmithyBuild builder;
+        private List<MockManifest> manifests;
+        private Map<String, String> expectedToContents;
+
+        public TestCaseBuilder name(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public TestCaseBuilder manifests(List<MockManifest> manifests) {
+            this.manifests = manifests;
+            return this;
+        }
+
+        public TestCaseBuilder builder(SmithyBuild builder) {
+            this.builder = builder;
+            return this;
+        }
+
+        public TestCaseBuilder expectedToContents(Map<String, String> expectedToContents) {
+            this.expectedToContents = expectedToContents;
+            return this;
+        }
+
+        public TestCase build() {
+            return new TestCase(this);
+        }
+    }
+
+    public static class JavaFileVisitor extends SimpleFileVisitor<Path> {
+        private final List<Path> javaFiles;
+
+        public JavaFileVisitor(List<Path> javaFiles) {
+            this.javaFiles = javaFiles;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+            if (Files.isRegularFile(file) && file.toString().endsWith(".java")) {
```

Comment:
Is there a reason to only load `.java` files? Codegen also produces property and SPI config files (`META-INF/services`).

Reply:
No particular reason, I can remove this bit.